core/capabilities/ccip/ccipsolana/commitcodec_test.go

```diff
@@ -153,3 +158,71 @@ func BenchmarkCommitPluginCodecV1_Encode_Decode(b *testing.B) {
 		require.Equal(b, rep, decodedReport)
 	}
 }
+
+func Test_DecodingCommitReport(t *testing.T) {
+	t.Run("decode on-chain commit report", func(t *testing.T) {
+		chainSel := cciptypes.ChainSelector(rand.Uint64())
+		minSeqNr := rand.Uint64()
+		maxSeqNr := minSeqNr + 10
+		onRampAddr, err := solanago.NewRandomPrivateKey()
+		require.NoError(t, err)
+
+		tokenSource := solanago.MustPublicKeyFromBase58("C8WSPj3yyus1YN3yNB6YA5zStYtbjQWtpmKadmvyUXq8")
+		tokenPrice := encodeBigIntToFixedLengthLE(big.NewInt(rand.Int63()), 28)
+		gasPrice := encodeBigIntToFixedLengthLE(big.NewInt(rand.Int63()), 28)
+		merkleRoot := utils.RandomBytes32()
+
+		tpu := []ccip_router.TokenPriceUpdate{
+			{
+				SourceToken: tokenSource,
+				UsdPerToken: [28]uint8(tokenPrice),
+			},
+		}
+
+		gpu := []ccip_router.GasPriceUpdate{
+			{UsdPerUnitGas: [28]uint8(gasPrice), DestChainSelector: uint64(chainSel)},
+			{UsdPerUnitGas: [28]uint8(gasPrice), DestChainSelector: uint64(chainSel)},
+			{UsdPerUnitGas: [28]uint8(gasPrice), DestChainSelector: uint64(chainSel)},
+		}
+
+		onChainReport := ccip_router.CommitInput{
+			MerkleRoot: ccip_router.MerkleRoot{
+				SourceChainSelector: uint64(chainSel),
+				OnRampAddress:       onRampAddr.PublicKey().Bytes(),
+				MinSeqNr:            minSeqNr,
+				MaxSeqNr:            maxSeqNr,
+				MerkleRoot:          merkleRoot,
+			},
+			PriceUpdates: ccip_router.PriceUpdates{
+				TokenPriceUpdates: tpu,
+				GasPriceUpdates:   gpu,
+			},
+		}
+
+		var buf bytes.Buffer
+		encoder := agbinary.NewBorshEncoder(&buf)
+		err = onChainReport.MarshalWithEncoder(encoder)
+		require.NoError(t, err)
+
+		commitCodec := NewCommitPluginCodecV1()
+		decode, err := commitCodec.Decode(testutils.Context(t), buf.Bytes())
```

Comment:
Could we also do the reverse for the same data?
I mean encode with commitCodec, and then decode with borsch decoder?